pkg/credentialprovider/config.go

```diff
@@ -40,28 +40,38 @@ type DockerConfigEntry struct {
 	Email    string
 }
 
-const (
-	dockerConfigFileLocation = ".dockercfg"
+var (
+	homeDirPath               = filepath.Join(os.Getenv("HOME") + ".dockercfg")
```

Comment:
I would not assume HOME is set, instead use the kubelet's getRootDir().  You'll have to plumb that in here.